ports/geckolib/glue.rs

```diff
@@ -3722,6 +3722,19 @@ pub extern "C" fn Servo_ProcessInvalidations(set: RawServoStyleSetBorrowed,
     let mut data = data.as_mut().map(|d| &mut **d);
 
     if let Some(ref mut data) = data {
```

Comment:
This needs to be a bit more convoluted. In particular, you only need to touch the parent if you have invalidated any sibling.

Also, this needs to use the special method added not long ago to avoid messing up with the flags in some edge cases, see https://bugzilla.mozilla.org/show_bug.cgi?id=1394935.

So this should become something like:

```rust
if result.invalidated_siblings() {
    let parent = element.traversal_parent().expect("How could we invalidate siblings without a common parent?");
    unsafe {
        parent.set_dirty_descendants();
        bindings::Gecko_NoteDirtySubtreeForInvalidation(parent.0);
    }
} else if result.invalidated_descendants() {
    unsafe { bindings::Gecko_NoteDirtySubtreeForInvalidation(element.0) };
} else if result.invalidated_self() {
    unsafe { bindings::Gecko_NoteDirtyElement(element.0) };
}
```